bundler/lib/bundler/templates/newgem/ext/newgem/extconf-c.rb.tt

```diff
@@ -2,4 +2,9 @@
 
 require "mkmf"
 
+# Makes all symbols private by default to avoid unintended conflict
+# with other gems. To explicitly export symbols you can use RUBY_FUNC_EXPORTED
+# selectively, or entirely remove this flag.
+$CFLAGS << " -fvisibility=hidden "
```

Comment:
Of course this option is compiler dependent.
Should be `append_cflags("-fvisibility=hidden")`.